relay-profiling/src/sample.rs

```diff
@@ -13,6 +13,12 @@ use crate::MAX_PROFILE_DURATION;
 
 const MAX_PROFILE_DURATION_NS: u64 = MAX_PROFILE_DURATION.as_nanos() as u64;
 
+#[derive(Debug)]
+struct Range {
+    start: usize,
+    end: usize,
+}
```

Comment:
You could use https://doc.rust-lang.org/std/ops/struct.Range.html for this. If you prefer to keep a right-inclusive range (`[a, b]`), I would rename its fields to `first` and `last`, to emphasize that its last element is included.